packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -219,7 +214,8 @@ export class TestObjectProvider {
             documentServiceFactory: this.documentServiceFactory,
             codeLoader,
```

Comment:
this isn't right now, we shouldn't expand it. we should create a new LoaderProps that we pass into the constructor, so
```
const realProps: ILoaderProps={
...loaderProps,
//then fill the the required/or modified here
}
```

Reply:
this should also allow us to get ride of detachedBlobStorage, but that can be phased

Reply:
I am a little confused. Are you saying that the code is incorrect now?
With what you are proposing, tests can pass in their own url resolvers, code loaders, etc. Do we want to allow that?

Reply:
https://github.com/microsoft/FluidFramework/pull/8879

Reply:
definitely. our test infra should give sane and simple defaults, but allow customizing just about anything. with this i can now easily inject a driver that fault injection, which was not possible before